ActiveMQ on Java 25

jeanouii opened the pull request:
No need to review, this is a temp PR I'll kill whenever I have more visibility on what now fails on Java 25

gemmellr commented on activemq-broker/src/main/java24/org/apache/activemq/broker/jmx/SubjectShim.java:
```diff
@@ -0,0 +1,35 @@
+package org.apache.activemq.broker.jmx;
+
+import javax.security.auth.Subject;
+import java.security.AccessController;
```
Looks like you have some stale commits from another PR.

Same months-old feedback [as on the other PR](https://github.com/apache/activemq/pull/1533#discussion_r2533507877), this is unused and shouldnt be here (and will break things when the SecurityManager APIs are eventually removed)

gemmellr commented on activemq-unit-tests/pom.xml:
```diff
@@ -465,6 +465,18 @@
+
+              This test needs to run as part of the integration test phase to ensure it's using the packaged version
+              We use failsafe instead which is a better choice for integration tests. See bellow the configuration.
+            -->
```
See earlier comment for potential route to avoid such config as this and the later inversion of it.

gemmellr reviewed (commented):
This should really be enforcing that Java 24+ is used for the release process since its required for this to be of any worth to end users.

An earlier PR is still open to require 21+ since Java 21+ wasnt used for the previous release when it should have been, though that PR (#1532) requires changes as its somewhat [out of date now](https://github.com/apache/activemq/pull/1532#discussion_r2727819960)

(I required 25 to release elsewhere, saves changing it later and noone should be really be using 24 now anyway)

jeanouii commented:
@gemmellr Thanks a lot for the detailed review. I grab back the branch and address the feedback.

jeanouii commented:
@gemmellr I've now added the enforcer plugin in the apache-release profile to make sure we release the MRJAR supporting Java 21 and Java 24.
I've removed the rule for the Maven version even though to be honest, I prefer it to be explicit like plugin versions or java compiler version. It's a personal preference and I'm fine if community prefers to yank it.
Let me know if that's ok, so I can create the issue in JIRA and then, squash update the commit messages to clean everything up

jbonofre commented on .github/workflows/ci-nightly.yml:
```diff
@@ -69,7 +69,7 @@ jobs:
           distribution: temurin
       - name: Test
-        run: mvn -B -e -fae test
+        run: mvn -B -e -fae verify
```
Any specific reason to use `verify` here more than `test` ? I'm just curious.
